verifier/src/evidence.rs

```diff
@@ -0,0 +1,334 @@
+        let quote_tcb_info = tcb_info_try_from_quote(&quote)?;
+        let tcb_info = TcbInfo::try_from(&tcb_info_raw)?;
+        let advisories = tcb_info.advisories(&quote_tcb_info)?;
+        Ok(Self {
+            quote,
+            tcb_info_raw,
+            advisories,
+        })
+    }
+}
+
+impl<'a, Q> Accessor<TcbInfoRaw<'a>> for Evidence<'a, Q> {
+    fn get(&self) -> TcbInfoRaw<'a> {
+        self.tcb_info_raw.clone()
+    }
+}
+
+impl<'a, Q: Clone> Accessor<Quote3<Q>> for Evidence<'a, Q> {
+    fn get(&self) -> Quote3<Q> {
+        self.quote.clone()
+    }
+}
+
+impl<'a, Q> Accessor<Advisories> for Evidence<'a, Q> {
+    fn get(&self) -> Advisories {
+        self.advisories.clone()
+    }
+}
+
+// TODO think this should go in tcb.rs of `mc-sgx-dcap-types`
+fn tcb_info_try_from_quote<Q: AsRef<[u8]>>(quote: &Quote3<Q>) -> Result<QuoteTcbInfo> {
+    let signature_data = quote.signature_data();
+    let certification_data = signature_data.certification_data();
+    let CertificationData::PckCertificateChain(pem_chain) = certification_data else {
+        return Err(Error::UnsupportedQuoteCertificationData);
+    };
+    let leaf_cert = pem_chain
+        .into_iter()
+        .next()
+        .ok_or(Error::UnsupportedQuoteCertificationData)?;
+    let certificate = Certificate::from_pem(leaf_cert)?;
+    Ok(QuoteTcbInfo::try_from(&certificate)?)
+}
+
+/// Macro to generate boilerplate for implementing [`Accessor`] for a field of
+/// the application enclave [`ReportBody`] in the evidence's [`Quote3`]
+///
+/// # Arguments
+/// * `field_type` - The type of the field in `ReportBody` to be accessed
+/// * `accessor_method` - The method on `ReportBody` that returns the field
+macro_rules! quote_application_report_body_field_accessor {
+    ($($field_type:ty, $accessor_method:ident;)*) => {$(
+        impl<'a, Q: AsRef<[u8]>> Accessor<$field_type> for Evidence<'a, Q> {
+            fn get(&self) -> $field_type {
+                self.quote.app_report_body().$accessor_method()
+            }
+        }
+    )*}
+}
+
+quote_application_report_body_field_accessor! {
+    Attributes, attributes;
+    ConfigId, config_id;
+    ConfigSvn, config_svn;
+    CpuSvn, cpu_svn;
+    ExtendedProductId, isv_extended_product_id;
+    FamilyId, isv_family_id;
+    IsvProductId, isv_product_id;
+    IsvSvn, isv_svn;
+    MiscellaneousSelect, miscellaneous_select;
+    MrEnclave, mr_enclave;
+    MrSigner, mr_signer;
+    ReportData, report_data;
+}
+
+#[cfg(test)]
+mod test {
+    extern crate alloc;
+    use super::*;
+    use crate::{
+        AdvisoriesVerifier, AdvisoryStatus, And, AttributesVerifier, MrSignerVerifier,
+        Quote3Verifier, TcbInfoRawVerifier, VerificationTreeDisplay, Verifier,
+    };
+    use alloc::format;
+    use core::mem;
+    use der::DateTime;
+    use mc_sgx_dcap_sys_types::{sgx_ql_ecdsa_sig_data_t, sgx_quote3_t};
+    use p256::ecdsa::VerifyingKey;
+
+    fn tcb_signing_key() -> VerifyingKey {
```

Comment:
This and the function below are duplicates from `tcb.rs` and `quote.rs`.
I'm ok with that. I think for unit tests duplicating small pieces of logic like this are more of a boon than a bane:
- It allows the tests to be independent from each other
- It prevents creating test support function with X inputs for each test that needs a slight tweak
- Building a common infrastructure of test support methods often gets treated as second or third class, degenerating into it's own maintenance burden.

for integration tests there is a better argument for test support functions in a common place.